### ISSUE TITLE
Building a target under certain circumstances

### ISSUE THREAD

**Issue by kirbyfan64** [vslavik/bakefile#70]
I have a situation where there are some targets in a Bakefile that should NOT be built by default. Any way to make it build will work, but I just need *some* way of having the target NOT be built by default. How should I do this?

**Comment by vadz** [comment]
It's not really possible to not build something by default in a MSVS solution, to the best of my knowledge, so I don't think you can do it. You can, of course, only build something for a particular toolset and so not build it at all for "gnu" for example. So maybe you could use a condition involving a variable defined to some non-default value in "gnu" case, e.g. as explained [here](file:///W:/dev/build/bakefile/github/docs/html/language.html#conditional-statements)?

**Comment by kirbyfan64** [comment]
@vadz Is there an example of that or something? Or was the `here` supposed to be a link somewhere?

**Comment by vadz** [comment]
Yes, it was -- except I gave a `file://` link to my local copy of the docs, surprisingly it didn't work here...

**Comment by kirbyfan64** [comment]
@vadz Well, that's weird... :O

The use case is that a file should only be built if Boost is present. I figured that I could just conditionally define the target via a `USE_BOOST` setting, but that doesn't work. I'm not sure if what you said would fit this too well...

Side note: in VS toolsets, is it possible to test configurations like:

```
if ($(config) == MyConfig) {
  sources { a.cpp }
} else {
  sources { b.cpp }
}
```

I'm guessing it's possible with the GNU ones, but I'm not sure about Visual Studio.
